diff --git a/src/Fl_Choice.cxx b/src/Fl_Choice.cxx
index e5a0276..a22c627 100644
--- a/src/Fl_Choice.cxx
+++ b/src/Fl_Choice.cxx
@@ -118,7 +118,7 @@ void Fl_Choice::draw() {
       fl_draw_shortcut = 2; // hack value to make '&' disappear
       l.draw(xx+3, yy, ww>6 ? ww-6 : 0, hh, FL_ALIGN_LEFT);
       fl_draw_shortcut = 0;
-      if ( Fl::focus() == this ) draw_focus(box(), xx, yy, ww, hh);
+      if ( Fl::focus() == this ) draw_focus(box(), xx-1, yy-2, ww+2, hh+4);
     }
     else {
       fl_draw_shortcut = 2; // hack value to make '&' disappear
diff --git a/src/Fl_Screen_Driver.cxx b/src/Fl_Screen_Driver.cxx
index 05461a2..641301f 100644
--- a/src/Fl_Screen_Driver.cxx
+++ b/src/Fl_Screen_Driver.cxx
@@ -478,37 +478,10 @@ int Fl_Screen_Driver::scale_handler(int event)
   if (!keyboard_screen_scaling) return 0;
   if ( event != FL_SHORTCUT || !Fl::event_command() ) return 0;
   enum {none, zoom_in, zoom_out, zoom_reset} zoom = none;
-  if (Fl::test_shortcut(FL_COMMAND+'+')) zoom = zoom_in;
-  else if (Fl::test_shortcut(FL_COMMAND+'-')) zoom = zoom_out;
-  else if (Fl::test_shortcut(FL_COMMAND+'0')) zoom = zoom_reset;
-
-  // Kludge to recognize shortcut FL_COMMAND+'+' without pressing SHIFT.
-  if (Fl::option(Fl::OPTION_SIMPLE_ZOOM_SHORTCUT)) {
-    if ((Fl::event_state() & (FL_META|FL_ALT|FL_CTRL|FL_SHIFT)) == FL_COMMAND) {
-      // We use Ctrl + key '=|+' for instance on US, UK, and FR keyboards.
-      // This works as expected on all keyboard layouts that have the '=' key in the
-      // lower and the '+' key in the upper position on the same key.
-      // This test would be "false positive" if a keyboard layout had the '=' key in
-      // the lower and any other key than '+' in the upper position!
-
-      if (Fl::event_key() == '=') zoom = zoom_in;
-
-      // Note: Fl::event_key() is often incorrect under X11 *if* the selected keyboard
-      // layout is *not* the primary one in the keyboard selection list, e.g. under Gnome.
-      // The observation is that Fl::event_key() is erroneously derived from the primary
-      // keyboard layout instead. This can be very confusing and I don't know why this
-      // happens. Albrecht-S, Oct 2024, on Debian 12 (Bookworm aka Stable as of now).
-
-      // Example: 0xfe51 ("dead_acute") is sent by the '=' key of the US layout if the
-      // primary layout is German. This *would* be the correct key value for the German
-      // keyboard layout but not for the US layout.
-      // The following statement would work around this for this very special case but
-      // this should IMHO not be done. A valid workaround by the user is to make the
-      // *used* layout the first in the keyboard layout selection list!
-
-      // else if (Fl::event_key() == 0xfe51) zoom = zoom_in; // dead_acute, see above
-    }
-  }
+  int key = Fl::event_key() & ~(FL_SHIFT+FL_COMMAND);
+  if (key == '=' || key == '+') zoom = zoom_in;
+  else if (key == '-') zoom = zoom_out;
+  else if (key == '0' || key == 0xE0) zoom = zoom_reset;
   if (zoom != none) {
     int i, count;
     if (Fl::grab()) return 0; // don't rescale when menu windows are on
@@ -543,9 +516,8 @@ int Fl_Screen_Driver::scale_handler(int event)
 #else
     // scaling factors for the standard GUI
     static float scaling_values[] = {
-      0.5f, 2.f/3, 0.8f, 0.9f, 1.0f,
-      1.1f, 1.2f, 4.f/3, 1.5f, 1.7f,
-      2.0f, 2.4f, 3.0f};
+            0.5f, 2.f/3, 0.8f, 0.9f, 1.00f, 1.25f, 1.50f, 2.00f, 2.50f, 3.00f, 4.00f, 5.00f
+    };
 #endif
     float f, old_f = screen_dr->scale(screen)/initial_scale;
     if (zoom == zoom_reset) f = 1;
diff --git a/src/Fl_Tabs.cxx b/src/Fl_Tabs.cxx
index 6d0f991..286d879 100644
--- a/src/Fl_Tabs.cxx
+++ b/src/Fl_Tabs.cxx
@@ -863,6 +863,7 @@ void Fl_Tabs::draw() {
   if (damage() & (FL_DAMAGE_ALL|FL_DAMAGE_EXPOSE|FL_DAMAGE_SCROLL))
   {
     // -- draw tabs background
+	if (damage() & FL_DAMAGE_ALL) {
     if (parent()) {
       Fl_Widget *p = parent();
       fl_push_clip(x(), tabs_y, w(), tabs_h);
@@ -876,6 +877,7 @@ void Fl_Tabs::draw() {
     } else {
       fl_rectf(x(), tabs_y, w(), tabs_h, color());
     }
+	}
 
     // -- draw selection border
     fl_push_clip(x(), selection_border_y, w(), selection_border_h);
@@ -1028,7 +1030,7 @@ void Fl_Tabs::draw_tab(int x1, int x2, int W, int H, Fl_Widget* o, int flags, in
 
     // Draw the focus box
     if (Fl::focus() == this && o->visible())
-      draw_focus(bt, x1, y(), W, H, bc);
+      draw_focus(bt, x1, y(), W, H-2, bc);
   } else {
     H = -H;
     H += dh;
diff --git a/src/Fl_win32.cxx b/src/Fl_win32.cxx
index e753973..eaf4944 100644
--- a/src/Fl_win32.cxx
+++ b/src/Fl_win32.cxx
@@ -561,7 +561,7 @@ void Fl_WinAPI_Screen_Driver::desktop_scale_factor() {
     if (r != S_OK) { dpiX = dpiY = 96; }
     dpi[ns][0] = float(dpiX);
     dpi[ns][1] = float(dpiY);
-    scale(ns, dpiX / 96.f);
+    scale(ns, 1.0f);
     // fprintf(LOG, "desktop_scale_factor ns=%d factor=%.2f dpi=%.1f\n", ns, scale(ns), dpi[ns][0]);
   }
   update_scaling_capability();
